app/src/main/java/com/google/samples/apps/sunflower/viewmodels/PlantDetailViewModel.kt

```diff
@@ -61,8 +62,8 @@ class PlantDetailViewModel(
         plant = plantRepository.getPlant(plantId)
     }
 
-    fun addPlantToGarden() {
-        viewModelScope.launch {
+    fun addPlantToGarden(): Job {
```

Comment:
Why do we need this? ViewModels/Presenters do not typically return values, correct?

Reply:
@dazza5000 thank you for your feedback.

I have declared it in the description of PR.

It is an OPEN/CLOSE of interface design. 

Actually, the return of `Job` means that the client codes can cancel the `addPlantToGarden` separately without call `viewmodel.coroutineScope.cancel`.

Imagine that there are more than one interface that use `coroutineScope` in this viewmoodel for async purpose, as long as  the `viewmodel.coroutineScope.cancel` is called, in order to cancel `addPlantToGarden`, all running interfaces which depend on `coroutineScope`  will be canceled as well, this is a drawback.